app/classifier/tasks/dropdown/index.cjsx

```diff
@@ -121,24 +120,30 @@ module.exports = React.createClass
   handleFocus: ->
     @menus[0].focus()
 
-  getOptions: (i, optionsKeys) ->
-    optionsKey = if optionsKeys.length then optionsKeys.join(';') else '*'
+  getOptionsKey: (i, optionsKey = '') ->
+    {selects} = @props.task
+    select = selects[i]
+    [condition] = selects.filter (filterSelect) => filterSelect.id is select.condition
+    conditionIndex = selects.indexOf(condition)
+    optionsKey = if optionsKey.length is 0 then @props.annotation.value[conditionIndex]?.value else "#{@props.annotation.value[conditionIndex]?.value};#{optionsKey}"
```

Comment:
How weird. `syncAnnotations` runs before mount so that selected options are properly set when the menus render. Keeping state in sync with props is a problem, though.